src/OrbitGl/App.cpp

```diff
@@ -1881,6 +1882,135 @@ Future<ErrorMessageOr<CanceledOr<std::filesystem::path>>> OrbitApp::RetrieveModu
   return chained_result_future;
 }
 
+Future<ErrorMessageOr<CanceledOr<std::filesystem::path>>> OrbitApp::RetrieveModuleItself(
+    const ModuleIdentifier& module_id, uint64_t module_file_size) {
+  ORBIT_SCOPE_FUNCTION;
+  ORBIT_CHECK(std::this_thread::get_id() == main_thread_id_);
+
+  auto find_locally_or_in_cache = [this, module_id,
+                                   module_file_size]() -> ErrorMessageOr<std::filesystem::path> {
+    if (absl::GetFlag(FLAGS_local)) {
```

Comment:
Why not find object in cache if `FLAGS_local`? 
Or are we doing something wrong now in `FindModuleLocallyImpl`? `FindModuleLocallyImpl` still check cache no matter `FLAGS_local` is enabled or not

Reply:
I think the idea behind this is: If we run locally, then the module actually needs to be available locally, if not, something weird is going on. And in this case it's a good idea to actually let the user now that 

There is actually another thing that could be considered. In some setups the `--local` flag is used in combination with a custom ssh tunnel. In this case the module path is not a local path and looking in the cache might make sense. 

I haven't fully thought this threw and am interested in opinions

Reply:
So what @antonrohr said.

> There is actually another thing that could be considered. In some setups the --local flag is used in combination with a custom ssh tunnel. In this case the module path is not a local path and looking in the cache might make sense.

This is something that affects the entire symbol loading, right? We already don't try to download a module to the cache if `--local` is used, or do we?

Reply:
if `--local` is used, we skip retrieving symbols from instance but might still look into Stadia and MC symbol stores and download them to cache.

Reply:
@vickyliu-go4it Indeed, but the point here is that we are not looking for a separate symbol file, but for the binary of the module itself.

Reply:
Here is one hard limitation: When `--local` is used, we do have a gRPC channel, but no ServiceDeployManager. Aka scp (sftp) downloads from the service to the client are not possible. `OrbitMainWindow::DownloadFileFromInstance` has this CHECK  `ORBIT_CHECK(std::holds_alternative<StadiaTarget>(...))`. (Maybe this can also be changed to return an `ErrorMessage` instead.

I think we should exhaust all possible searching options. Aka also search in the cache in this case.

Reply:
Alright, sounds good. Done. Please have a look.

Reply:
LGTM!